From 3e534bce65f24858b3dac041a35cdd973258e94f Mon Sep 17 00:00:00 2001
From: Andy Ning <andy.ning@windriver.com>
Date: Thu, 28 Aug 2025 09:50:04 -0400
Subject: [PATCH 01/10] Not ask for user to input keyring password

With this patch, keyring no longer ask user for password when
encrypt/decrypt passwords stored in keyring.

This patch also updated keyring storage file permisson to 644.

Signed-off-by: Andy Ning <andy.ning@windriver.com>
---
 keyrings/alt/file.py      | 30 ++++++++++++++++++++++--------
 keyrings/alt/file_base.py |  2 +-
 2 files changed, 23 insertions(+), 9 deletions(-)

diff --git a/keyrings/alt/file_base.py b/keyrings/alt/file_base.py
index aa70a49..6fcd005 100644
--- a/keyrings/alt/file_base.py
+++ b/keyrings/alt/file_base.py
@@ -163,7 +163,7 @@ class Keyring(FileBacked, KeyringBackend):
             # create the file without group/world permissions
             with open(self.file_path, 'w', encoding='utf-8'):
                 pass
-            user_read_write = 0o600
+            user_read_write = 0o644
             os.chmod(self.file_path, user_read_write)
 
     def delete_password(self, service, username):
diff --git a/keyrings/alt/file.py b/keyrings/alt/file.py
index 8d0bfed..b406c3d 100644
--- a/keyrings/alt/file.py
+++ b/keyrings/alt/file.py
@@ -63,11 +63,18 @@ class Encrypted:
 
     def _get_new_password(self):
         while True:
-            password = getpass.getpass("Please set a password for your new keyring: ")
-            confirm = getpass.getpass('Please confirm the password: ')
-            if password != confirm:  # pragma: no cover
-                sys.stderr.write("Error: Your passwords didn't match\n")
-                continue
+#****************************************************************
+# Forging the Keyring password to allow automation and still keep
+# the password encoded. TODO to be revisited when Barbican keyring
+# Will be used with the complete PKI solution
+#****************************************************************
+#            password = getpass.getpass("Please set a password for your new keyring: ")
+#            confirm = getpass.getpass('Please confirm the password: ')
+#            if password != confirm:  # pragma: no cover
+#                sys.stderr.write("Error: Your passwords didn't match\n")
+#                continue
+            password =  "Please set a password for your new keyring: "
+
             if '' == password.strip():  # pragma: no cover
                 # forbid the blank password
                 sys.stderr.write("Error: blank passwords aren't allowed.\n")
@@ -179,9 +186,16 @@ class EncryptedKeyring(Encrypted, Keyring):
         Unlock this keyring by getting the password for the keyring from the
         user.
         """
-        self.keyring_key = getpass.getpass(
-            'Please enter password for encrypted keyring: '
-        )
+#****************************************************************
+# Forging the Keyring password to allow automation and still keep
+# the password encoded. TODO to be revisited when Barbican keyring
+# Will be used with the complete PKI solution
+#****************************************************************
+#        self.keyring_key = getpass.getpass(
+#            'Please enter password for encrypted keyring: '
+#        )
+        self.keyring_key = "Please set a password for your new keyring: "
+
         try:
             ref_pw = self.get_password('keyring-setting', 'password reference')
             assert ref_pw == 'password reference value'
-- 
2.34.1

